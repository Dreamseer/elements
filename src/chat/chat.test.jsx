/**
 * Chat Test
 * @author Marc Görtz <https://marcgoertz.de/>
 */

import React from 'react';
import renderer from 'react-test-renderer';
import Chat from '.';

const mockHistory = [
  {
    content: 'O HAI! 👋🏼',
  },
];

const mockFullHistory = [
  {
    name: 'Costello',
    content: 'Look, you gotta first baseman?',
    timestamp: new Date(1938, 2, 24, 12, 11, 10),
  },
  {
    name: 'Abbott',
    content: 'Certainly.',
    timestamp: new Date(1938, 2, 24, 12, 11, 35),
    isMe: true,
  },
  {
    name: 'Costello',
    content: 'Who’s playing first?',
    timestamp: new Date(1938, 2, 24, 12, 11, 50),
  },
  {
    name: 'Abbott',
    content: 'That’s right.',
    timestamp: new Date(1938, 2, 24, 12, 12, 4),
    isMe: true,
  },
  { content: 'Costello becomes exasperated.', isAction: true },
  {
    name: 'Costello',
    content:
      'When you pay off the first baseman every month, who gets the money?',
    timestamp: new Date(1938, 2, 24, 12, 12, 39),
  },
  {
    name: 'Abbott',
    content: 'Every dollar of it.',
    timestamp: new Date(1938, 2, 24, 12, 12, 50),
    isMe: true,
  },
  {
    name: 'Abbott',
    content: '😁',
    timestamp: new Date(1938, 2, 24, 12, 13, 0),
    isMe: true,
    isEmoji: true,
  },
];

describe('<Chat />', () => {
  it('renders no chat by default', () => {
    const component = renderer.create(<Chat />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders a basic chat', () => {
    const component = renderer.create(<Chat history={mockHistory} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders a full-featured chat', () => {
    const component = renderer.create(<Chat history={mockFullHistory} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
