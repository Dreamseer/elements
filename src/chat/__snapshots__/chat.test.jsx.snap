// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> renders a basic chat 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  list-style: none;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto, Helvetica,Arial,sans-serif,'Apple Color Emoji','Segoe UI Emoji', 'Segoe UI Symbol';
  line-height: 1.25;
}

.c1 {
  position: relative;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  padding: 0.5rem;
  border-radius: 0.75rem;
  outline: 0;
  max-width: 60%;
  color: black;
  background-color: gainsboro;
  box-sizing: border-box;
}

.c1 + li {
  margin-top: 0.5rem;
}

.c1:active time {
  display: block;
}

.c2 {
  display: block;
  quotes: none;
}

<ol
  className="c0"
>
  <li
    className="c1"
    tabIndex={null}
  >
    <q
      className="c2"
    >
      O HAI! üëãüèº
    </q>
  </li>
</ol>
`;

exports[`<Chat /> renders a full-featured chat 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  list-style: none;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto, Helvetica,Arial,sans-serif,'Apple Color Emoji','Segoe UI Emoji', 'Segoe UI Symbol';
  line-height: 1.25;
}

.c7 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0.5rem 0;
  text-align: center;
  max-width: 80%;
  font-size: 0.75rem;
  font-weight: bold;
  color: gray;
}

.c7 i {
  font-style: normal;
}

.c1 {
  position: relative;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  padding: 0.5rem;
  border-radius: 0.75rem;
  outline: 0;
  max-width: 60%;
  color: black;
  background-color: gainsboro;
  box-sizing: border-box;
}

.c1 + li {
  margin-top: 0.5rem;
}

.c1:active time {
  display: block;
}

.c5 {
  position: relative;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  padding: 0.5rem;
  border-radius: 0.75rem;
  outline: 0;
  max-width: 60%;
  color: white;
  background-color: dodgerblue;
  box-sizing: border-box;
}

.c5 + li {
  margin-top: 0.5rem;
}

.c5:active time {
  display: block;
}

.c8 {
  position: relative;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  padding: 0.5rem;
  border-radius: 0.75rem;
  outline: 0;
  max-width: 60%;
  color: white;
  background-color: dodgerblue;
  box-sizing: border-box;
  padding: 0;
  font-size: 4em;
  background-color: transparent;
}

.c8 + li {
  margin-top: 0.5rem;
}

.c8:active time {
  display: block;
}

.c2 {
  display: block;
  margin-bottom: 0.25em;
  font-size: 0.75rem;
  font-weight: bold;
}

.c9 {
  display: none;
  margin-bottom: 0.25em;
  font-size: 0.75rem;
  font-weight: bold;
}

.c3 {
  display: block;
  quotes: none;
}

.c4 {
  display: none;
  position: absolute;
  top: 100%;
  left: 0.666667em;
  margin-top: 0.25em;
  text-align: left;
  font-size: 0.75rem;
  color: gray;
  opacity: 0.75;
  white-space: nowrap;
}

.c6 {
  display: none;
  position: absolute;
  top: 100%;
  right: 0.666667em;
  margin-top: 0.25em;
  text-align: right;
  font-size: 0.75rem;
  color: gray;
  opacity: 0.75;
  white-space: nowrap;
}

<ol
  className="c0"
>
  <li
    className="c1"
    tabIndex="0"
  >
    <span
      className="c2"
    >
      Costello
    </span>
    <q
      className="c3"
    >
      Look, you gotta first baseman?
    </q>
    <time
      className="c4"
      dateTime="1938-03-24"
    >
      03/24/1938
    </time>
  </li>
  <li
    className="c5"
    tabIndex="0"
  >
    <span
      className="c2"
    >
      Abbott
    </span>
    <q
      className="c3"
    >
      Certainly.
    </q>
    <time
      className="c6"
      dateTime="1938-03-24"
    >
      03/24/1938
    </time>
  </li>
  <li
    className="c1"
    tabIndex="0"
  >
    <span
      className="c2"
    >
      Costello
    </span>
    <q
      className="c3"
    >
      Who‚Äôs playing first?
    </q>
    <time
      className="c4"
      dateTime="1938-03-24"
    >
      03/24/1938
    </time>
  </li>
  <li
    className="c5"
    tabIndex="0"
  >
    <span
      className="c2"
    >
      Abbott
    </span>
    <q
      className="c3"
    >
      That‚Äôs right.
    </q>
    <time
      className="c6"
      dateTime="1938-03-24"
    >
      03/24/1938
    </time>
  </li>
  <li
    className="c7"
  >
    Costello becomes exasperated.
  </li>
  <li
    className="c1"
    tabIndex="0"
  >
    <span
      className="c2"
    >
      Costello
    </span>
    <q
      className="c3"
    >
      When you pay off the first baseman every month, who gets the money?
    </q>
    <time
      className="c4"
      dateTime="1938-03-24"
    >
      03/24/1938
    </time>
  </li>
  <li
    className="c5"
    tabIndex="0"
  >
    <span
      className="c2"
    >
      Abbott
    </span>
    <q
      className="c3"
    >
      Every dollar of it.
    </q>
    <time
      className="c6"
      dateTime="1938-03-24"
    >
      03/24/1938
    </time>
  </li>
  <li
    className="c8"
    tabIndex="0"
  >
    <span
      className="c9"
      hidden={true}
    >
      Abbott
    </span>
    <q
      className="c3"
    >
      üòÅ
    </q>
    <time
      className="c6"
      dateTime="1938-03-24"
    >
      03/24/1938
    </time>
  </li>
</ol>
`;

exports[`<Chat /> renders no chat by default 1`] = `null`;
